java-version/accelerator.yaml

```diff
@@ -27,10 +27,34 @@ engine:
         regex:
           pattern: "sourceCompatibility = .*"
           with: "'sourceCompatibility = ''' + #javaVersion + ''''"
-    - include: [ "config/workload.yaml" ]
+    - include: [ "build.gradle.kts" ]
       chain:
       - type: ReplaceText
-        condition: "#javaVersion == '17'"
-        substitutions:
-          - text: "spec:"
-            with: "'spec:\n  build:\n    env:\n    - name: BP_JVM_VERSION\n      value: \"17\"'"
+        regex:
+          pattern: "(?<unmodified>JavaVersion\\.VERSION_)(\\d*)"
+          with: "'${unmodified}' + #javaVersion.replace('.', '_')"
+    - condition: "#files.contentsOf('config/workload.yaml').get().contains('- name: BP_JVM_VERSION')"
+      include: [ "config/workload.yaml" ]
+      chain:
+      - condition: "#javaVersion == '1.8'"
+        include: [ "config/workload.yaml" ]
```

Comment:
This is extremely brittle (and unreadable). Should leverage a YAML oriented OpenRewrite recipe instead, and if it doesn't suit your needs, we need to try to make it better.

Reply:
Condition checking provided java version is replaced with a variable which is calculated at the beginning of the 'engine' section.
YAML transformation is still done through text replacement as OpenRewrite doesn't support manipulation of elements of a collection.

Reply:
Thanks @ericbottard. It works now. I probably had a syntax mistake. OpenRewrite-based solution is pushed.

Reply:
This is now resolved